src/Arcus.BackgroundJobs.AzureActiveDirectory/AzureApplication.cs

```diff
@@ -32,14 +32,14 @@ public ApplicationWithExpiredAndAboutToExpireSecrets(string name, Guid keyId, Da
         public string Name { get; }
 
         /// <summary>
-        /// Gets the id of the clientsecret.
+        /// Gets the id of the client secret.
         /// </summary>
-        public Guid KeyId { get; }
+        public Guid? KeyId { get; }
 
         /// <summary>
-        /// Gets the end datetime of the clientsecret.
+        /// Gets the end datetime of the client secret.
         /// </summary>
-        public DateTime EndDateTime { get; }
+        public DateTimeOffset? EndDateTime { get; }
```

Comment:
Changed this to nullable, as we don't actually have to fail if it's not present.